### PR TITLE
ci: add license check using github actions

### PR DESCRIPTION
**Pull request by nashif**
Use Giithub for license checks.

### CODE REVIEW

**Review by galak — COMMENTED** [review]
nit, commit message: 'giiiiiit'hub ;)

**Review by galak — APPROVED** [review]

File: .github/workflows/license_check.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Scancode
+
+on: [pull_request]
+
+jobs:
+  scancode_job:
+    runs-on: ubuntu-latest
+    name: Scan code for licenses
+    steps:
+    - uses: actions/checkout@v1
```
Comment (galak) [inline .github/workflows/license_check.yml]: I moved to `checkout@v2`.  Seems a bit smarter/faster.

File: .github/workflows/license_check.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    - name: Artifact Upload
+      uses: actions/upload-artifact@v1
+      with:
+        name: scancode
+        path: ./artifacts
+
+    - name: Verify
+      run: |
+        test ! -s ./artifacts/report.txt || (cat ./artifacts/report.txt && exit 1 )
+
```
Comment (galak) [inline .github/workflows/license_check.yml]: Nit: whitespace.

File: .github/license_config.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+license:
+  main: apache-2.0
+  report_missing: true
+  category: Permissive
+copyright:
+  check: true
+exclude:
+  extensions:
+    - yml
```
Comment (galak) [inline .github/license_config.yml]: Do we really want to exclude `yml/yaml`?  Is it scancode or some other check that flags YAML files needed copyright & license headers?

File: .github/workflows/license_check.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+name: Scancode
+
+on: [pull_request]
```
Comment (galak) [inline .github/workflows/license_check.yml]: I added `[pull_request, push]`. I know we don't ever have push's but figured its better to have it in here.

Comment (nashif) [reply .github/workflows/license_check.yml]: this action only acts on changed files, on push we will need to scan the whole tree which is fine I guess, but I will need to update a few things to make this work.

Comment (galak) [reply .github/workflows/license_check.yml]: Ah, that makes sense.

File: .github/license_config.yml (hunk 3 above)
Comment (nashif) [reply .github/license_config.yml]: every file is scanned, if we add yaml files we will have to change 100s of them in the tree and add license/copyright.

File: .github/workflows/license_check.yml (hunk 1 above)
Comment (nashif) [reply .github/workflows/license_check.yml]: ok, will check it out :)

**Comment by nashif** [comment]
DNM while the action is adapted and tested for the main tree.

File: .github/license_config.yml (hunk 3 above)
Comment (galak) [reply .github/license_config.yml]: Aren't we doing that today?  I know I got a message about adding license/copyright to my .github/workflow/<FOO>.yml.

Even you have it in this PR:

```
.github/license_config.yml missing copyright.
.github/workflows/license_check.yml missing license.
.github/workflows/license_check.yml missing copyright.
```

Comment (nashif) [reply .github/license_config.yml]: > Even you have it in this PR:

yeah, we exclude yaml, but not yml in ci-tools

Comment (nashif) [reply .github/license_config.yml]: > yeah, we exclude yaml, but not yml in ci-tools, which means that something is wrong in the script, will take a look.

actually no, the above is from the ci-tools checker which does not exclude .yml, the action does things correctly.
In any case, I would not be changing policy now, first get this in, then the config can be changed based on needs.

**Pull request merged**